spleeter/audio/__init__.py

```diff
@@ -10,6 +10,42 @@
     - Waveform convertion and transforming functions.
 """
 
+from enum import Enum
+
+# pyright: reportMissingImports=false
+# pylint: disable=import-error
+import tensorflow as tf
+# pylint: enable=import-error
+
 __email__ = 'spleeter@deezer.com'
 __author__ = 'Deezer Research'
 __license__ = 'MIT License'
+
+
+class Codec(str, Enum):
+    """ Enumeration of supported audio codec. """
+
+    WAV: str = 'wav'
+    MP3: str = 'mp3'
+    OGG: str = 'ogg'
+    M4A: str = 'm4a'
+    WMA: str = 'wma'
+    FLAC: str = 'flac'
```

Comment:
while you are at it, i would propose to "decouple" the codec from the container and reflect this change in the API

Reply:
Hey thanks for you comment !

Actually this is an issue we addressed internally on other projects as we are using same Adapter pattern with FFMPEG, and one of our team member noticed some edge case were the previous implementation returned wrong values which lead to FFMPEG returning empty arrays.

We will perform a full review of this PR once it is finished, I would recommend you to wait before to back port this patch to your adapter just to be sure but I think you should indeed do it at the end :).

Concerning the codec vs container issue this is indeed maybe the right place to setup such thing or at least to start thinking about it. This is quite a heavy PR as we are performing lot of changes without breaking compatibility. But switching to Typer actually will make the process of adding or updating command line options a lot more easier so once it is done we should be able to address this problem in a second iteration probably. We will discuss that with the team this week and let you know.

Reply:
@Faylixe sounds good.

Maybe related: I cover an additional ffmpeg time issue with very small numbers in my adapter: https://github.com/faroit/stempeg/blob/master/stempeg/read.py#L193-L197

not sure if `str(dt.timedelta(seconds=duration))` makes this unnecessary, but better safe than sorry ;-)

Reply:
Nice, well I think the standard python library should handle precision but to be tested to ensure that with the new version.

Reply:
i tested it in spleeter, and things are good!